examples/pattern-example.yaml

```diff
@@ -0,0 +1,86 @@
+id: https://w3id.org/linkml/examples/pattern
+title: Pattern Example
+name: pattern-example
+description: This demonstrates the use of patterns
+license: https://creativecommons.org/publicdomain/zero/1.0/
+
+prefixes:
+  linkml: https://w3id.org/linkml/
+  ex: https://w3id.org/linkml/examples/patterns/
+  sh: https://w3id.org/shacl/
+
+default_prefix: ex
+default_range: string
+
+default_curi_maps:
+    - semweb_context
+
+emit_prefixes:
+  - linkml
+  - rdf
+  - rdfs
+  - xsd
+  - owl
+
+imports:
+  - linkml:types
+
+# constant/parameter settings
+settings:
+  float: "\\d+[\\.\\d+]"
+  unit: "\\S+"
+  unit.length: "(centimeter|meter|inch)"
+  unit.weight: "(kg|g|lbs|stone)"
+  email: "\\S+@\\S+{\\.\\w}+"
+  
+#==================================
+# Classes                         #
+#==================================
+
+classes:
+  PersonInfo:
+    slots:
+      - id
+      - name
+      - height
+      - email
+
+#==================================
+# Slots                           #
+#==================================
+
+slots:
+  id:
+    identifier: true
+    range: string
+    pattern: "^P\\d{7}"    ## simple pattern. Note {}s are NOT interpolated
+  name:
+    range: string
+  email:
+    range: EmailString
+  height:
+    range: string
+    structured_pattern:
+      pattern: "{float} {unit.length}"
```

Comment:
Technical question, can a structured_pattern pattern have a mixture of {setting} and re expressions, like  "{float}\s*{unit.length}" ?

Reply:
Side note, it seems like for javascript based programming, at runtime load of a linkml spec, each given "pattern" can be converted into a compiled regular expression by expanding it out and then re.compile() ?  I've done this for very efficient runtime validation of data.

Reply:
Q1: I had envisioned yes, and that most implementations would interpolate directly to a regex.

JS: yes, that's the idea. And stay tuned re JS and linkml...